tools/generate_image.sh

```diff
@@ -14,51 +16,55 @@ Usage: $(basename $0) image_path image_partitions build_dir [version.json_file]
 ==usage
 }
 
-if [[ ( $# -lt 3 )  || ( $# -gt 8 ) ]]; then
+if [ $# -lt 3 ] || [ $# -gt 8 ]; then
 	echo "Error! Invalid argument count $# args: $*"
 	usage
-	exit -1
+	exit 1
 fi
 
-IMAGE_NAME=$(realpath $1)
-PRODUCT_NAME=$(basename -s .img $1)
-IMAGE_PARTITIONS=$(realpath $2)
+L_GIT_DIR=$(git rev-parse --show-toplevel)
+if [ ! -d "$L_GIT_DIR" ]; then L_GIT_DIR=$(readlink -f "$(dirname "$0")"/..); fi
+
+. "$L_GIT_DIR"/tools/version_check.sh
+. "$L_GIT_DIR"/tools/locate_bins.sh
+require_commands sfdisk mke2fs mtools mformat mcopy mmd truncate sync realpath basename
+
+IMAGE_NAME=$($call_realpath "$1")
+PRODUCT_NAME=$($call_basename -s .img "$1")
+IMAGE_PARTITIONS=$($call_realpath "$2")
+LUTS=""
+VERSION_FILE=""
+BIN_FILE=""
+UPDATER_FILE=""
 
-SYSROOT=$(realpath $3)
-LUTS=$4
-VERSION_FILE=$5
-BIN_FILE=$6
-UPDATER_FILE=$7
+SYSROOT=$($call_realpath "$3")
 
+if [ $# -gt 3 ]; then
+	LUTS="$4"
+fi
+if [ $# -gt 4 ]; then
+	VERSION_FILE="$5"
+fi
+if [ $# -gt 5 ]; then
+	BIN_FILE="$6"
+fi
+if [ $# -gt 6 ]; then
+	UPDATER_FILE="$7"
+fi
```

Comment:
`set -u` requires `$4`/`$5`/`$6`/`$7` only be used if they are defined and therefore we must define variables as being empty strings beforehand.